src/sidebar/components/ShareDialog/FileInput.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  FileGenericIcon,
+  IconButton,
+  Input,
+  InputGroup,
+} from '@hypothesis/frontend-shared';
+import type { MutableRef } from 'preact/hooks';
+import { useId } from 'preact/hooks';
+
+export type FileInputProps = {
+  onFileSelected: (file: File) => void;
+  fileInputRef: MutableRef<HTMLInputElement | null>;
+  disabled?: boolean;
+};
+
+export default function FileInput({
+  onFileSelected,
+  disabled,
+  fileInputRef,
+}: FileInputProps) {
+  const inputId = useId();
+
+  return (
+    <>
+      <input
+        ref={fileInputRef}
+        accept=".json"
+        type="file"
+        disabled={disabled}
+        className="invisible absolute w-0 h-0"
+        data-testid="file-input"
+        onChange={e => {
+          const files = (e.target as HTMLInputElement)!.files;
+          if (files !== null && files.length > 0) {
+            onFileSelected(files[0]);
+          }
+        }}
+      />
+      <label htmlFor={inputId}>Select Hypothesis export file:</label>
+      <InputGroup>
+        <Input
+          id={inputId}
+          onClick={() => fileInputRef.current?.click()}
+          readonly
+          disabled={disabled}
+          value={fileInputRef.current?.files?.[0]?.name ?? 'Select a file'}
```

Comment:
I know, but since there are other things in the chain that are already nullable, it doesn't make much difference.

Basically `fileInputRef.current?.files![0]?.name` is mostly equivalent to `fileInputRef.current?.files?.[0]?.name` here.

Reply:
It does convey a different intent though. If I use a `?` it suggests a situation where I expect the value might be missing, as opposed to making it clear that I have some knowledge that TypeScript doesn't about what the expected values at runtime will be.